core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -1930,8 +1930,13 @@ public RelBuilder filter(Iterable<CorrelationId> variablesSet,
     if (config.simplify()) {
       conjunctionPredicates = simplifier.simplifyFilterPredicates(predicates);
     } else {
+      List<RexNode> simplified = new ArrayList<>();
```

Comment:
minor improvement here:
```List<RexNode> simplified = new ArrayList<>(predicates.size);```

Reply:
predicates is an iterable, so you don't gain much, since you may need to traverse it twice